src/Core/Maintenance/System/Command/SystemUpdateFinishCommand.php

```diff
@@ -33,6 +35,17 @@ public function __construct(ContainerInterface $container)
         $this->container = $container;
     }
 
+    protected function configure(): void
```

Comment:
please don't type hint as void because of the BC, please add a `@deprecated tag:v6.5.0 - reason-return-type-change - The return type will change to void`

Reply:
@keulinho 

1. Can you check discussion regarding this in previous PR **?**
   - The point is if I do not add the void then another check fails, also most of your commands added `: void` once they implemented configure and didn't before. 
   - See comment: https://github.com/shopware/platform/pull/2379#issuecomment-1063133457
   - and comment: https://github.com/shopware/platform/pull/2379#issuecomment-1063814129
   - If` you still want I can change it though and ignore the other failing check.
2. Personally, I am against adding `@deprecated` here because using deprecated means the method itself is deprecated, this is an issue with IDEs and various lint tools as the say the method is deprecated and warn about it. I actually noticed this on other places where you use deprecated but it is incorrect in such scenarios and might trigger lint warnings or errors. Personally I think a TODO or similar comment would make more sense. 
    - But if this is some guideline I can do so, but as mentioned using deprecated is actually wrong here, as IDEs will show them accordingly if used and linter (e.g. qodana) will also report them as issues.

Reply:
Ah i understand, but 
```
/*
 * @deprecated tag:v6.5.0 - reason-return-type-change - The return type will change to void
 *
 * @return void
 */
```
Should be completely backwards compatible and fine for the static analysis too. Can you try that? If that does not work I'm fine with excluding it for the BC-checker.

But the BC-check is there for a reason, and we should not and exclusions there without really good reasons to do so.

Regarding your second point, it's true that we use @deprecated not only in cases where the method/class will be removed, but also in all cases where some other breaking change will be applied to the method in the next major version. We decided to do so and don't use a simple @todo or custom annotation because of the IDE support, as we wanted to let external know if some BC happens to methods they are using. Honestly i think official support for an annotation like that is missing, and until then we probably stick to the @deprecated

Reply:
@keulinho Done. I guess with the phpdoc / deprecated info, the other failing pipelines should not trigger but we will see soon

Reply:
Oh wow, the cs-fixer thinks it's safe to add the return type, so the cs-fixer and the bc-check contradict here... 

In that case please ignore my last comment and revert the latest commit, and ignore the break over the bc-excludes

Reply:
@keulinho Done

Reply:
Yep exactly that is the scenario that will lead to a break, however a break is a break and we should not break something without first trying it in a non-breaking way ;)

In this case the cs-fixer works in some kind of unexpected way, as normally the cs-fixer just runs on new code, so adding `void` there is no break, however there is edge case when we add a implementation of a method that was previously inherited from the parent class. 

I'll keep an eye on that specific problem, so we can fix the cs-fixer if it will pop up again, but as already said for this case it is ok to do the break